CellProfiler_pipelines/README.md

```diff
@@ -0,0 +1,288 @@
+The type of SQLite file wanted for this project is one that has a table for every object, I changed the parameter to `One table per object type`.
+
+**Note:** Having `One table per object type` set as the parameter will cause a warning to pop up that says this parameter will cause the file outputted to not be used in CellProfiler Analyst (CPA) properly. 
+Since we are not using CPA, this warning can be ignored.
+
+## Step 1: Setup CellProfiler Environment
+
+### Step 1a: Create Environment
+ 
+ I used this [.yml file format](https://github.com/CellProfiler/CellProfiler/wiki/Conda-Installation) from the Wiki. 
+ I updated the file and the Wiki page so that it would pip install the newest verison of CellProfiler instead of the older version that it had before.
+
+ ```sh
+# Run this command in terminal to create the conda environment for CellProfiler
+conda env create -f download_cellprofiler.yml
+```
+
+### Step 1b: Activate Environment
+
+```sh
+# Run this command in terminal to activate the conda environment for CellProfiler
+conda activate cp4
+```
+
+## Step 2: Open CellProfiler
+
+```sh
+# Run this command in terminal to start CellProfiler
+cellprofiler
+```
+
+## Step 3: Run `illum.cpproj` and `analysis.cpproj`
+
+`illum.cpproj` will be run first before running `analysis.cpproj`.
+The same steps in order below will be followed for each project.
+
+When CellProfiler is open, you will drag the CP project file from your file explorer into the box on the CellProfiler interface, as shown below:
+
+![CellProfiler_fig1.png](Example_Images/CellProfiler_fig1.png)
+
+After the project is loaded in, it will look like the image below:
+
+![CellProfiler_pipeline.png](Example_Images/CellProfiler_pipeline.png)
+
+Make sure that the `Images` module contains all of the NF1 images like the above image shows before proceeding.
+Because I saved the pipeline with the images from the direcotry from my computer, the images will need to be deleted and readded into the `Images` module to reflect the correct path in your machine.
+
+Before running the pipeline, change the "Output file location" in the `ExportToDatabase` module to the desired location for the SQLite file.
+
+To run the illumination correction pipeline, press `Analyze Images` at the bottom right.
+
+**Note:** If you do not want to see the pop ups from each module in the pipeline, you can turn the `eye symbol` off, like seen in the image above for all modules underneath `IdentifyTeritaryObjects`. 
+This is done by just pressing that symbol to turn it on or off depending on what modules you want to view.
+All of the segmentation modules are turned on as to see the how CellProfiler is segmenting each of the images as pipeline runs so you can stop the pipeline if you see something wrong in the segmentation.
+
+### Runtime
+
+I ran these pipelines on using a computer running Pop!_OS 20.04 LTS 64-bit with a AMD® Ryzen 7 3700x 8-core processor × 16 and NVIDIA graphics card.
```

Comment:
Does CP take advantage of the GPU?

Also, you probably should add more specs about the GPU (RTX 3090), and maybe include that you have 64 GB DDR4 RAM 🤷

Reply:
I do not think it does based on my Google search.

There is a paper from 2018 where they try to implement GPU into CellProfiler but the base CellProfiler seems to only use CPU. Here is the link to the paper: https://ieeexplore.ieee.org/document/8621271